File: Practice/AllDomains/Tutorials/30DaysOfCode/Day25RunningTimeandComplexity.js
/*	author:@shivkrthakur  */
function processData(input) {
    //Enter your code here
    var inputArr = input.split('\n').map(Number);
    for(var i = 1; i < inputArr.length; i++){
      if(isPrime(inputArr[i])) console.log('Prime');
      else console.log('Not prime');
    }
}

function isPrime(n){
    if(n > 1 && n % 2 > 0){
        var sqrt = Math.floor(Math.sqrt(n));
        for(var i = 3; i <= sqrt; i += 2){
            if(n % i == 0) return false;
        }
        return true;
    }
    else return (n === 2);
}
process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day25RunningTimeandComplexity.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;

class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int testCasesCount = Convert.ToInt32(Console.ReadLine());
        int[] testCases = new int[testCasesCount];

        for(int i = 0; i < testCasesCount; i++)
        {
            int number = Convert.ToInt32(Console.ReadLine());
            testCases[i] = number;
        }

        for(int i = 0; i < testCasesCount; i++)
        {
            if(IsPrime(testCases[i]))
                Console.WriteLine("Prime");
            else
                Console.WriteLine("Not prime");
        }
    }

    static bool IsPrime(int number)
    {
        if(number <= 0 || number == 1)
            return false;

        int nSqrt = Convert.ToInt32(Math.Sqrt(number));

        for(int x=2; x <= nSqrt; x++ )
        {
            if(IsPrime(x))
            {
                if((number % x) == 0)
                    return false;
            }
        }
        return true;
    }
}
